app/models/miq_alert.rb

```diff
@@ -72,6 +73,14 @@ def set_responds_to_events
     self.responds_to_events = events unless events.nil?
   end
 
+  def validate_delay_next_evaluation
+    next_frequency = (options || {}).fetch_path(:notifications, :delay_next_evaluation)
+    evaluation_method = expression[:eval_method] if expression.kind_of?(Hash)
+    if !evaluation_method.nil? && evaluation_method == "dwh_generic" && !next_frequency.nil? && next_frequency != 0
+      errors.add(:notifications, "Datawarehouse alerts must have a 0 notification frequency")
```

Comment:
Yes those names are much better, unfortunately that is the name in the UI.
At this point we might ask to change that. see https://github.com/ManageIQ/manageiq-ui-classic/pull/678#issuecomment-288057359